lib/config/folderconfiguration.go

```diff
@@ -20,8 +20,8 @@ type FolderConfiguration struct {
 	ID                    string                      `xml:"id,attr" json:"id"`
 	Label                 string                      `xml:"label,attr" json:"label"`
 	RawPath               string                      `xml:"path,attr" json:"path"`
+	Type                  string                      `xml:"type,attr" json:"type"`
```

Comment:
Everything looks good to me, except the attribute type here. I don't think this should be a `string`, it should be a `FolderType` similar to `PullOrder`, which is an enum that knows how to serialize itself to/from text. Then the default value of zero equals `readwrite` and we don't run the risk of it suddenly having the value `"banana"` or `""` and panicing somewhere.